File: 1-Patron/php/GestorDeUltimate.php
<?php
class GestorDeUltimate
{
  private $ultimate;

  function __construct($ultimate)
  {
    $this->$ultimate = $ultimate;
  }

  function subirDeNivel($incremento)
  {
    $this->ulimate->incNivel($incremento);
  }

  function generarUltimate($nombre, $dano, $nivel)
  {
    return new Ultimate($nombre, $dano, $nivel);
  }
}
?>

File: 1-Patron/Java/GestordeUltimate.java

package masterwholeproxy;

public class GestordeUltimate {
    private Ultimate ultimate;
    
    public GestordeUltimate(Ultimate ultimate){
        this.ultimate = ultimate;
    }
    public String subirDeNivel(int incremento){
        this.ultimate.incNivel(incremento);
        return ("se incremento el ultimate "+ incremento+ " puntos");
    }
    public Ultimate generarUltimate(String nombre, int dano, int nivel){
        return new Ultimate(nombre,dano,nivel);
    }
}
